javascript/blogs/quick-sort/quick-sort.md

```diff
@@ -0,0 +1,68 @@
+# Code Fellows 401 JavaScript: Code Challenge for Class 28 (13 January 2020)
+
+## Author: Nathan Rhead Cox
+
+## Prompt
+
+Review the pseudocode below, then trace the algorithm by stepping through the process with the provided sample array. Document your explanation by creating a blog article that shows the step-by-step output after each iteration through some sort of visual. Once you are done with your article, code a working, tested implementation of Merge Sort based on the pseudocode provided.
+
+```
+ALGORITHM QuickSort(arr, left, right)
+    if left < right
+        // Partition the array by setting the position of the pivot value 
+        DEFINE position <-- Partition(arr, left, right)
+        // Sort the left
+        QuickSort(arr, left, position - 1)
+        // Sort the right
+        QuickSort(arr, position + 1, right)
+
+ALGORITHM Partition(arr, left, right)
+    // set a pivot value as a point of reference
+    DEFINE pivot <-- arr[right]
+    // create a variable to track the largest index of numbers lower than the defined pivot
+    DEFINE low <-- left - 1
+    for i <- left to right do
+        if arr[i] <= pivot
+            low++
+            Swap(arr, i, low)
+
+     // place the value of the pivot location in the middle.
+     // all numbers smaller than the pivot are on the left, larger on the right. 
+     Swap(arr, right, low + 1)
+    // return the pivot index point
+     return low + 1
+
+ALGORITHM Swap(arr, i, low)
+    DEFINE temp;
+    temp <-- arr[i]
+    arr[i] <-- arr[low]
+    arr[low] <-- temp
+```
+
+## Algorithm
+
+Input: array = [8,4,23,42,16,15].
+Output: array = [4,8,15,16,23,42].
+
+Concept: 
+
+
+
+Slide 1 represents 
+
+![Slide 1 Visual](../assets/code-challenge-28-slide1.png)
+
+Slide two represents 
+
+![Slide 2 Visual](../assets/code-challenge-28-slide2.png)
+
+Conclusion:
+
+## Efficiency
+
+- Time = O(n^2): The comparison happens n * (n-1) number of times, making the algorithm n-squared for time.
+- Space: O(1): The array is sorted in place, so no new array is created, making the space used constant.
+
+## References
+
+
```

Comment:
You're off to a great start, completely understandable with the internet/power outage issues, come back to this when you have time!